src/firefly/java/edu/caltech/ipac/util/download/UriRef.java

```diff
@@ -92,12 +92,7 @@ static private UriRef makeString(String uriStr) {
     }
 
     private static URL makeURLFromStr(String urlStr) {
-        if (urlStr == null) return null;
-        try {
-            return new URI(urlStr.trim()).toURL();
-        } catch (URISyntaxException | MalformedURLException ignore) {
-            return null;
-        }
+        return Util.Try.it(() -> new URI(urlStr).toURL()).getOrElse((URL)null);
```

Comment:
I think we the trim and null check made sense (at least the trim, I understand the `Try` util will take care of the null exception. So suggested change : 

```
if (urlStr == null) return null;
return Util.Try.it(() -> new URI(urlStr.trim()).toURL()).getOrElse((URL)null);
```

You can skip the first line, but might make sense to keep the `.trim()` in there.